app/waf/src/controllers/adc.controller.ts

```diff
@@ -732,74 +685,154 @@ export class AdcController extends BaseController {
       });
     }
   }
+
+  private async onboarding(adc: Adc, addon: AddonReqValues): Promise<void> {
+    try {
+      this.logger.debug('start to do onbarding');
+      await this.serialize(adc, {status: AdcState.ONBOARDING});
+
+      let doMgr = await OnboardingManager.instanlize(this.wafapp);
+      let doBody = await doMgr.assembleDo(adc, {onboarding: true});
+      let doId = await doMgr.onboarding(doBody);
+
+      await checkAndWait(() => doMgr.isDone(doId), 240)
+        .then(() =>
```

Comment:
seems to check whether DO is DONE has 2 steps ,  IMHO, it's better if we could do this consistently for other parts, meaning  all checks for DO also go in to the AdcStateEntry.check() method, which i think it's the intended design in that AdcStateCtrlr class.  But I don't mind if we leave it until next time.

Reply:
You are right. It's OK to putting them into state controller. 

For this case, `isDone` is to check the API execution is over. 
If it returns non-200, check will be terminated immediately. left it here may be earlier to discover error when happens.